src/transformers/models/layoutlmv3/tokenization_layoutlmv3_fast.py

```diff
@@ -640,18 +639,23 @@ def _batch_encode_plus(
                 else:
                     original_index = batch_index
                 labels_example = []
+                previous_token_empty = False
                 for id, offset, word_id in zip(
                     sanitized_tokens["input_ids"][batch_index],
                     sanitized_tokens["offset_mapping"][batch_index],
                     sanitized_encodings[batch_index].word_ids,
                 ):
                     if word_id is not None:
                         if self.only_label_first_subword:
-                            if offset[0] == 0:
+                            if offset[0] == 0 and not previous_token_empty:
                                 # Use the real label id for the first token of the word, and padding ids for the remaining tokens
                                 labels_example.append(word_labels[original_index][word_id])
                             else:
                                 labels_example.append(self.pad_token_label)
+                            if offset == (0, 0):
```

Comment:
```suggestion
                            if self.decode(id) == "":
```
I'm not sure offset == (0,0) is the right way to check this, maybe this is a safer option

Reply:
Yes I believe it is equivalent, although slower than simply comparing a tuple to (0, 0) it is probably safer and resilient to future changes.
It works because special tokens are checked line 648 with `word_id is not None`. Thus an offset of (0, 0) cannot be a special token at this position in code.

Reply:
I just checked for LayoutLMv3TokenizerFast, `tok.decode(1437) == " "`, where 1437 is the id of the "Ġ" token.

Reply:
Ok interesting. When working on `UdopTokenizerFast` (UDOP is a new model for which I'll open a PR soon), I had to use `self.decode(id) == ""`, cause it didn't work with `offset == (0, 0)`. UDOP has the same vocabulary as T5, which is based on SentencePiece.

When testing out the following with offset == (0,0):
```
from transformers import UdopTokenizerFast

words = ['a', 'weirdly', 'test', 'hello']
boxes = [[1,2,3,4] for _ in range(len(words))]
labels = [1, 2, 3, 4]

tokenizer = UdopTokenizerFast.from_pretrained("nielsr/udop-large")

encoding = tokenizer(words, boxes=boxes, word_labels=labels)

for id, label in zip(encoding.input_ids, encoding.labels):
    print(tokenizer.decode([id]), label)
```
it gives me:
```
 1
a -100
weird 2
ly -100
test 3
hello 4
</s> -100
```
(I used "weirdly" just to make sure we get multiple tokens). Interestingly it splits the word "a" into 2 tokens: an empty token and "a". I also printed (id, offset, word_id):
```
 (0, 1) 0
a (0, 1) 0
weird (0, 5) 1
ly (5, 7) 1
test (0, 4) 2
hello (0, 5) 3
```
and in this case the empty token has offset (0, 1), which explains why offset == (0,0) didn't work.

Reply:
This is probably not related to the current issue but your tokenizer produces weird (pun intended) offsets mapping.
```
 (0, 1)
a (0, 1)
weird (0, 5)
ly (5, 7)
```
We should be able to derive word length from the `offset_mapping`, ie "weirdly" is of length `(5-0) + (7-5) = 7`. But this does not hold anymore with this empty token not being assigned `(0, 0)` offset.